include/mega/types.h

```diff
@@ -209,6 +209,43 @@ typedef enum ErrorCodes
     API_EBUSINESSPASTDUE = -28      ///< Business account expired
 } error;
 
+class Error
+{
+public:
+    Error(error err = API_EINTERNAL)
+    {
+        mError = err;
+        mExtraErrorInfo = false;
+    }
+
+    void setError(error err)
+    {
+        mError = err;
+        mExtraErrorInfo = false;
+    }
+
+    void setErrorWithExtraInfo(error err)
+    {
+        mError = err;
+        mExtraErrorInfo = true;
```

Comment:
If `err` appear in json, extra info should be provided. We avoid to assign value twice to `mExtraErrorInfo = true;`. But this is not very important

Reply:
Documentation says:
> 2) If yes, is the extra information available? (It should be assumed that there may be cases it is not provided, just in case).
It's not clear whether the error without extra info will be received with the old fashion or the new format, but just in case I think it's more correct to assume there's extra info only when there's extra info, since the new format might become the standard in the future for reporting any error for any format ;)